mesonbuild/backend/ninjabackend.py

```diff
@@ -576,7 +576,15 @@ def detect_prefix(out):
 
         raise MesonException(f'Could not determine vs dep dependency prefix string. output: {stderr} {stdout}')
 
-    def generate(self):
+    def generate(
+        self,
+        capture: bool = False,
+        captured_compile_args_per_buildtype_and_target: dict = None
+    ) -> T.Optional[dict]:
```

Comment:
Meson does not use black-style formatting, in particular, please do NOT include the closing parenthesis for a function definition at the same indentation level as the preceding "def".

Also this parameter name seems a bit on the long side... I suspect that is the only reason you broke it onto multiple lines to begin with.

Reply:
On your point about closed parenthesese indentation:  I find this -
```
def somefunc(
    self,
    something,
    another,
    yet_more,
    and_finally) -> None
    self.start.doing_some_stuff_with(something)
    ...
```
makes the first line of the method's body get lost with the with the params.  While this -
```
def somefunc(
    self,
    something,
    another,
    yet_more,
    and_finally
) -> None
    self.start.doing_some_stuff_with(something)
    ...

```
gives a much clearest distinction between multiple params and function body.
Perhaps you prefer this -
```
def somefunc(
    self,
    something,
    another,
    yet_more,
    and_finally
    ) -> None
    self.start.doing_some_stuff_with(something)
    ...
```
which I find a little clearer than the first snippet above, but not quite as good as the second, which you're not keen on, it seems.
What's it to be?

Reply:
Also, regarding the long name:  Whenever it's a choice between clarity and conciseness for naming, I always favour clarity.

For want of a better example -
```
world_records_by_sex_and_sport
```
makes it clearer (when compared to a variable named just 'world_records') that you could expect to use it like -
```
record = world_records_by_sex_and_event['male']['100m']
```
which I think marginally accelerates comprehension for the reader, especially if someone mistakenly misused it, like -
```
record = world_records_by_sex_and_event['100m']['male']
```
... Anyone reading this can, I think, more more easily see the mistake in order of applying keys here, while this would be much harder to misuse or spot the mistake if we were only using `world_records`.

If you insist, I'll can change all uses of the long name to just `captured_compile_args`, but I just wanted to put forward the case for my preference in this case.

Reply:
The style that is used everywhere else is to have the closing parenthesis and return type on the same line as the last argument. Whether we should change it is a separate discussion but for now this MR should follow that style.

Reply:
> On your point about closed parenthesese indentation:  I find this -
> ```
> def somefunc(
>     self,
>     something,
>     another,
>     yet_more,
>     and_finally) -> None
>     self.start.doing_some_stuff_with(something)
>     ...
> ```
> makes the first line of the method's body get lost with the with the params.  While this -


But really it should be this:
```
def somefunc(self,
             something,
             another,
             yet_more,
             and_finally) -> None:
    self.start.doing_some_stuff_with(something)
    ...
```


> [...]
> which I find a little clearer than the first snippet above, but not quite as good as the second, which you're not keen on, it seems.

There's quite a bit of your third suggestion used in the codebase, which is to say, having the closing paren on a new line but still indented more than four spaces to line up with the opening paren.

Also somewhat common:
```python
def somefunc(self, something, another, yet_more,
             will_it_end, and_finally) -> None:
    self.start.doing_some_stuff_with(something)

Reply:
In terms of what I would call style bugs that must be fixed, functions currently use the style that arguments are aligned with the opening paren, and the close paren is either inline with the last argument or on its own line matching the indent of arguments.

This is how I *believe* most or all current code is styled.

How many arguments to use per line is typically left to the taste of the code author.

Whether to give the close paren its own line is also typically left to the taste of the code author with a bias towards doing so specifically when the return annotation takes up a large width.

Reply:
Apart from shortening the long variable name (I wasn't sure whether this was a show-stopper for you, in light of my arguments on this point above), this is done in [51b0700](https://github.com/mesonbuild/meson/pull/11049/commits/51b070087670a477acc298a58915e310e0ece191).

If you really do prefer, let me know and I'll also shorten `captured_compile_args_per_buildtype_and_target` to `captured_compile_args` under protest 😄